### PR TITLE
Split RUC LSM into land and sea ice parts

### PR DESCRIPTION
**Pull request by tanyasmirnova**
The title above should be a 1 line short summary of the pull request (i.e. what the project the PR represents is intended to do).

Enter a description of this PR. This should include why this PR was created, and what it does.
1. The module_sf_ruclsm.F is plit into module_ruc_land.F90 and module_ruc_ice.F90 to be called separately from driver_lsm and driver_seaice. The Makefile is changed appropriately.
2. mpas_atmphys_driver_lsm.F - removed all remaining stuff related to sea ice. 
3. mpas_atmphys_driver_seaice.F - added RUC LSM variables and a call to ruc_ice.
4. 2-m RUC diagnostics is moved to driver_seaice (as in MPAS 8.1). 
5. To obtain identical results with the old version, CQS array was added to MYNN surface layer scheme. This array is needed to compute a factor CQS/CQS2 for linear interpolation to 2 m. If we compute it in the seaice driver using mavail (soil moisture availability) as it was before in the driver_lsm, the results will not be identical because mavail is recomputed in ruc_land, and 2-m diagnostics is called after from driver_seaice after call to ruc_ice.
6. Added debugging prints to driver_lsm, driver_seaice, ruc_land and ruc_ice.

Testing and relations to other Pull Requests should be added as subsequent comments.
1. Testing was preformed on 15-km: results are very close.
2. Testing on 3-km CONUS domain for 9 May 2024. 

See the below examples for more information.
https://github.com/MPAS-Dev/MPAS/pull/930
https://github.com/MPAS-Dev/MPAS/pull/931

### CODE REVIEW

File: src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F [hunk 1]
```diff
@@ -1519,7 +1527,7 @@ subroutine driver_sfclayer(itimestep,configs,mesh,diag_physics,sfc_input,its,ite
                    rho3d    = rho_p      , dz8w      = dz_p     , cp       = cp          , &
                    g        = gravity    , rovcp     = rcp      , R        = R_d         , &
                    xlv      = xlv        , chs       = chs_p    , chs2     = chs2_p      , &
-                   cqs2     = cqs2_p     , cpm       = cpm_p    , znt      = znt_p       , &
+                   cqs = cqs_p,cqs2     = cqs2_p     , cpm       = cpm_p    , znt      = znt_p       , &
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F]: Laura will despise this messing up her formatting.

File: src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F [hunk 2]
```diff
@@ -1550,7 +1558,7 @@ subroutine driver_sfclayer(itimestep,configs,mesh,diag_physics,sfc_input,its,ite
                       rho3d    = rho_p      , dz8w      = dz_p     , cp       = cp          , &
                       g        = gravity    , rovcp     = rcp      , R        = R_d         , &
                       xlv      = xlv        , chs       = chs_sea  , chs2     = chs2_sea    , &
-                      cqs2     = cqs2_sea   , cpm       = cpm_sea  , znt      = znt_sea     , &
+                      cqs=cqs_sea,cqs2     = cqs2_sea   , cpm       = cpm_sea  , znt      = znt_sea     , &
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F]: same here

**Review by joeolson42 — APPROVED** [review]
Just some formatting issues that would likely be enough to prevent a merging back into MMM's repo someday.

File: src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F (hunk 1 above)
Comment (tanyasmirnova) [reply src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F]: @joeolson42 you are right, Joe. I fixed it now. Thank you for bringing this to my attention.

**Comment by AndersJensen-NOAA** [comment]
@tanyasmirnova: Did you run tests with both debug=True and debug=False?

**Comment by tanyasmirnova** [comment]
> @tanyasmirnova: Did you run tests with both debug=True and debug=False?

@AndersJensen-NOAA Yes, I did.

**Review by AndersJensen-NOAA — APPROVED** [review]

**Review by mdtoyNOAA — APPROVED** [review]

**Pull request merged**